build-tools/packages/build-cli/src/library/release.ts

```diff
@@ -70,38 +65,63 @@ export interface ReleaseRanges {
 	 * A tilde version range. Equivalent to patch.
 	 */
 	tilde: string;
+
+	/**
+	 * A legacy compatibility range that is configurable per release group.
+	 * This range extends beyond standard version ranges and lies between major and minor versions.
+	 * Exceeding this range indicates compatibility differences.
+	 */
+	legacyCompat: string;
 }
 
 /**
  * Get the release ranges for a version string.
  *
  * @param version - The version.
+ * @param legacyCompatInterval - The multiple of minor versions to use for calculating the next version in the range.
  * @param scheme - If provided, this version scheme will be used. Otherwise the scheme will be detected from the
  * version.
  * @returns The {@link ReleaseRanges} for a version string
  */
-export const getRanges = (version: ReleaseVersion, scheme?: VersionScheme): ReleaseRanges => {
+export const getRanges = (
+	version: ReleaseVersion,
+	legacyCompatInterval: LegacyCompatInterval,
+	releaseGroup: ReleaseGroup | undefined,
+	scheme?: VersionScheme,
+): ReleaseRanges => {
 	const schemeToUse = scheme ?? detectVersionScheme(version);
+
 	return schemeToUse === "internal"
 		? {
 				patch: getVersionRange(version, "patch"),
 				minor: getVersionRange(version, "minor"),
 				tilde: getVersionRange(version, "~"),
 				caret: getVersionRange(version, "^"),
+				legacyCompat: getLegacyCompatVersionRange(
+					version,
+					legacyCompatInterval,
+					releaseGroup,
+					schemeToUse,
+				),
 			}
```

Comment:
Should we support legacy compat for internal scheme? If so, this will need to follow that internal pattern. (getVersionRange would probably need updated)
@tylerbutler, opinion?
I am fine with skipping internal support. Assuming we need to emit a legacyCompat entry here, I guess fallback to `~`.?

Reply:
We should add a comment here. As I was writing out my thoughts, I think `^` is probably more appropriate. Given our past history with internals I don't see a need to be cautious with limiting to minor. And nice that it matches our general fallback.
I would say something like. legacyCompat is not currently supported for internal scheme. Fallback to same major compatibility range.